### ISSUE TITLE
output file stops working with --type-check flag

### ISSUE THREAD

**Issue by rhysburnie** [palantir/tslint#2539]
Hi,

This works fine
```
tslint -p tsconfig.json -c tslint.json 'src/**/*.tsx' -o lintResults.txt
```

But when I use --type-check it fails to write file and logs in stdout instead
```
tslint --type-check -p tsconfig.json -c tslint.json 'src/**/*.tsx' -o lintResults.txt
```
This is a code base written by a third party my company hired and its really bad.
I need to analyse it.

I'm trying to get a count of errors when type checked...
The first results with 13665 errors it must be far more with type check but so many I can't scroll to the top of them in terminal.

Is there a way to get it to write files in type-check mode?

Cheers

**Comment by ajafff** [comment]
Currently your files are not linted when there are type check errors. These errors are printed to stderr.
Instead of using `-o` you can just write stderr to a file with `tslint ... 2> lintResult.txt`

**Comment by rhysburnie** [comment]
OK thanks.

Also is there a way to output the actual rule name in the log?

I wrote a script to count the different errors but because some errors use specific values from the code base like: 

`The class property 'context' must be marked either 'private', 'public', or 'protected'`

I cant collate them (without first knowing what they are) for example bellow the lines with **The class property** are the same rule.

```
[13664]: TOTAL ERRORS

[176]: statements are not aligned
[69]: file should end with a newline
[1]: space indentation expected
[189]: A maximum of 1 class per file is allowed
[1]: The class property 'context' must be marked either 'private', 'public', or 'protected'
[1]: The class property 'refs' must be marked either 'private', 'public', or 'protected'
[1]: The class property 'state' must be marked either 'private', 'public', or 'protected'
[1]: The class property 'props' must be marked either 'private', 'public', or 'protected'
[1]: The class property 'contextTypes' must be marked either 'private', 'public', or 'protected'
[243]: The class method 'render' must be marked either 'private', 'public', or 'protected'
[1]: Declaration of public static field not allowed after declaration of public instance field. Instead, this should come at the beginning of the class/interface.
[709]: Forbidden 'var' keyword, use 'let' or 'const' instead
...
```

**Comment by ajafff** [comment]
use `-t verbose` to use the verbose formatter which includes the rule name

**Comment by domoritz** [comment]
In this file

```ts
import {buildModel} from '../src/compile/common';
import {FacetModel} from '../src/compile/facet';
import {LayerModel} from '../src/compile/layer';
import {Model} from '../src/compile/model';
import {UnitModel} from '../src/compile/unit';
import {initConfig} from '../src/config';
import {ExtendedSpec, FacetSpec, LayerSpec, normalize, TopLevel, UnitSpec} from '../src/spec';

export function parseModel(inputSpec: TopLevel<ExtendedSpec>): Model {
  const spec = normalize(inputSpec);
  return buildModel(spec, null, '', initConfig(inputSpec.config));
}

export function parseUnitModel(spec: TopLevel<UnitSpec>) {
  return new UnitModel(spec, null, '', initConfig(spec.config));
}

export function parseLayerModel(spec: TopLevel<LayerSpec>) {
  return new LayerModel(spec, null, '', initConfig(spec.config));
}

export function parseFacetModel(spec: TopLevel<FacetSpec>) {
  return new FacetModel(spec, null, '', initConfig(spec.config));
}
```

I get these errors:

```
ERROR: test/util.ts[7, 9]: 'ExtendedSpec' is declared but never used.
ERROR: test/util.ts[7, 23]: 'FacetSpec' is declared but never used.
ERROR: test/util.ts[7, 34]: 'LayerSpec' is declared but never used.
ERROR: test/util.ts[7, 66]: 'UnitSpec' is declared but never used.
```

It looks like tslint thinks that variables in generics are not used.
